### ISSUE TITLE
Need comfirm fix in handling "Seek And Destroy" mission

### ISSUE THREAD

**Issue by universe-11** [naev/naev#979]
Naev v0.7.0 (x86_64-pc-linux-gnu)
 git HEAD at a2d958ab9
BFD backtrace catching enabled.
 Sea of Darkness

scenario:
I finished a "Seek And Destroy" mission and not landing in that authority at once to claim it. I went on to do other missions. After a while, I want to claim it and go back there and suddenly the same target show up again. I missed to catch the target and the computer told me I did not finish the mission and had to do it again. 

console output:
Warning: [misn_runFunc] Mission 'Seek And Destoy' -> 'hail': [string "dat/missions/neutral/seek_n_destroy.lua"]:202: bad argument #1 to 'ipairs' (table expected, got nil)
stack traceback:
[C]: in function 'ipairs'
[string "dat/missions/neutral/seek_n_destroy.lua"]:202: in function 'elt_inlist'
[string "dat/missions/neutral/seek_n_destroy.lua"]:343: in function <[string "dat/missions/neutral/seek_n_destroy.lua"]:340>
Warning: [hook_runMisn] Hook [hail] '59' -> 'hail' failed

**Comment by nenau-again** [comment]
Mm. @onpon4 's fix has probably solved the warning.
But I'm unsure if the bug is actually related to this warning... So best imo is to wait and see if someone gets it again on the current master. I'll try to reproduce it if I can get some time.
By the way, this mission is very new, so if you have any feedback, I'd be happy to read it.

**Comment by onpon4** [comment]
OK, so just as a clarification: when we talk about mission claims, that's a thing missions do when they are created to prevent mission conflicts, not something a player does.

I'm honestly not entirely sure what you're saying happened. All I fully understand is that the Lua warning (which would have led to the Lua code no longer being executed) is caused by a particular table being uninitialized (hence: "table expected, got nil"). So what I did is add an initialization of said table earlier in the script.

It could also be that the table needs a "__save" entry; I didn't look thoroughly enough in the mission code to know for sure (was busy with other things). Perhaps I'll take a closer look later today.

**Comment by onpon4** [comment]
I've investigated this a bit further and I've found that the most likely cause of problems is that the mission used the "jumpin" hook to set things up, and didn't account for takeoff into the target system. So it would have been possible to not get the proper system setup when taking off into the target system after loading a save. I've accounted for this by switching it from a jumpin hook to an enter hook in my bugfix branch (#977).
